proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -260,7 +260,7 @@ message Message {
   // file_path is the normalized path of the file containing the message.
   string file_path = 5;
   bool is_map_entry = 6;
-  repeated Field fields = 7;
+  repeated MessageField fields = 7;
```

Comment:
Having reviewed the [specification](https://developers.google.com/protocol-buffers/docs/overview#updating) and [encoding](https://developers.google.com/protocol-buffers/docs/encoding#embedded), I think embedding the `oneof` in a nested message actually make this wire incompatible. We're introducing new field numbers for the `MessageField` (those referenced in its `oneof` field), and those will be encoded in the wire format.

From the specification,
> Changing a single optional value into a member of a new oneof is safe and binary compatible. Moving multiple optional fields into a new oneof may be safe if you are sure that no code sets more than one at a time. Moving any fields into an existing oneof is not safe.

Unfortunately, we can't flatten out `repeated` values in a `oneof` at this level since it's not supported, i.e. 
```protobuf
// This isn't valid.
oneof message_field {
  repeated Field fields = 7;
  repeated Oneof oneofs = 10;
}
```

Anyway, this might have already been thought out and I might be missing some context, but I just wanted to call out that this will break compatibility between the BSR and the documentation front-end if they aren't deployed alongside one another.

Reply:
I think we expect breaking changes for v1alpha1, but yes this is not wire compatible.

Reply:
Yeah, that makes sense -- originally I had encoded the `oneofs` as a separate field in the `Message` struct, but as per the comment below, we wanted to try and retain the field ordering for documentation, but this could be handled on the frontend (especially since this isn't wire compatible 😅 ).